bindings/rust/s2n-tls/src/renegotiate.rs

```diff
@@ -59,21 +176,111 @@ impl RenegotiateCallback for RenegotiateResponse {
     }
 }
 
+#[derive(Debug, PartialEq, Copy, Clone, Default)]
+pub(crate) struct RenegotiateState {
+    needs_handshake: bool,
+    needs_wipe: bool,
+    send_pending: bool,
+}
+
 impl Connection {
+    fn schedule_renegotiate(&mut self) {
+        let mut state = self.renegotiate_state_mut();
+        if !state.needs_handshake {
+            state.needs_handshake = true;
+            state.needs_wipe = true;
+        }
+    }
+
+    fn is_renegotiating(&self) -> bool {
+        self.renegotiate_state().needs_handshake
+    }
```

Comment:
It still wouldn't be practical with concurrent reads/writes bc renegotiation will fail on a pending / blocked send. We can of course make it public later if necessary (and I considered another mode where that's all we do-- instead of automatic renegotiation we just surface the "is_renegotiating" state for you to make integration slightly easier). But I don't see any real benefit right now.

Reply:
Oh, another point: the primary use case for this right now is for it to be usable via s2n-tls-tokio or s2n-tls-hyper by JUST setting some config. If the application has to call a new method during IO in order to use it, that won't work when s2n-tls-tokio or s2n-tls-hyper is managing IO.

Reply:
Makes sense!